### PR TITLE
LPD-31931 - Address ERC gaps in Headless APIs for carts

### PR DESCRIPTION
**Pull request by matyaswollner**
https://liferay.atlassian.net/browse/LPD-31931

### CODE REVIEW

**Comment by matyaswollner** [comment]
ci:test:sf

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/cart/resource/v1_0/test/CartResourceTest.java [hunk 1]
```diff
@@ -74,6 +81,26 @@ public void setUp() throws Exception {
...
+			"XY", "XYZ", true, true, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+			RandomTestUtil.nextDouble(), true, true, false, _serviceContext);
+
+		_region = _regionLocalService.addRegion(
+			_country.getCountryId(), true, RandomTestUtil.randomString(),
+			RandomTestUtil.nextDouble(), RandomTestUtil.randomString(),
+			_serviceContext);
+
+		_cartAddress = _addressLocalService.addAddress(
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-test/src/testIntegration/java/com/liferay/headless/commerce/delivery/cart/resource/v1_0/test/CartResourceTest.java:95]: can you move this directly in the tests like we are doing in _testPatchCartByExternalReferenceCodeWithMoreExternalReferenceCodes with the Address randomAddress? And call it randomAddress

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by matyaswollner** [comment]
ci:test:sf

**Review by smottal — APPROVED** [review]

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartResourceImpl.java [hunk 2]
```diff
@@ -958,16 +958,42 @@ private void _updateOrder(CommerceOrder commerceOrder, Cart cart)
...
+
+			if (commerceAddress == null) {
+				billingAddressId = commerceOrder.getBillingAddressId();
+			}
+			else {
+				billingAddressId = commerceAddress.getCommerceAddressId();
+			}
+		}
+
+		long shippingAddressId = cart.getShippingAddressId();
```
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartResourceImpl.java:977]: missing GetterUtil.getLong(

**Comment by smottal** [comment]
Resent here: https://github.com/liferay-commerce/liferay-portal/pull/5300

**Pull request closed without merging**